meson.build

```diff
@@ -0,0 +1,116 @@
+project('dosbox-staging', 'c', 'cpp',
+        default_options : ['cpp_std=c++14', 'b_ndebug=if-release'],
+        version : '0.77.0')
+
```

Comment:
Suggest we include a lower-bound Meson version requirement to give users a friendly message to upgrade - and prevent users experiencing Python backtraces or undefined syntax.

Reply:
`meson_version : '>= 0.55.0'`, Although I'm not sure how low we can go in practice without incrementally testing older versions.

Reply:
For our C files, we can add:
`add_project_arguments('-std=c99', language : 'c')`

To ensure our code stays strictly c99-compliant and doesn't pull in GNU'isms, without rippling this requirement down into subprojects and dependencies.

Reply:
Nice! I did not know we can embed this info in the build file :)

Ubuntu 18.04 is at 0.45.1 and I don't want to go any older ATM (for Ubuntu 16.04 I don't care any more, it's almost EOL). But we'll revise this after next Debian Stable release.

Reply:
> For our C files, we can add:
> `add_project_arguments('-std=c99', language : 'c')`
> 
> To ensure our code stays strictly c99-compliant and doesn't pull in GNU'isms, without rippling this requirement down into subprojects and dependencies.

add_project_arguments('-std=c99', language : 'c')

This breaks dr_mp3 library :( `-std=c99` turns on `-ansi`, when `-ansi` is enabled dr_mp3 redefine `inline` to `__inline__` to work with c89 and in result inlining is broken in our c99 code.

Reply:
Oh - strange! 

Is meson (incorrectly) turning on `-ansi` when it sees `-std=c99`, or is that GCC doing this pairing? (Just wonder if we should submit a bug to Meson).

We have `AC_PROG_CC_C99` in `configure.ac`, but when I `make V=1`, I'm not seeing `-std=c99` being passed to the compiler for `.c` files.. so maybe that line is useless?

``` text
ccache gcc -DHAVE_CONFIG_H -I. -I../../..  -I../../../include -I/usr/include/SDL2 -D_REENTRANT   -Wall -pipe -fdiagnostics-color=auto -fpch-preprocess -DNDEBUG -O3 -fstrict-aliasing -fno-signed-zeros -fno-trapping-math -fassociative-math -frename-registers -ffunction-sections -fdata-sections -mfpmath=sse -msse4.2 -pthread -Wpedantic -Wall -DUSE_OPUS  -Wall -pipe -fdiagnostics-color=auto -fpch-preprocess -DNDEBUG -O3 -fstrict-aliasing -fno-signed-zeros -fno-trapping-math -fassociative-math -frename-registers -ffunction-sections -fdata-sections -mfpmath=sse -msse4.2 -pthread -MT libdecoders_a-vorbis.o -MD -MP -MF .deps/libdecoders_a-vorbis.Tpo -c -o libdecoders_a-vorbis.o `test -f 'vorbis.c' || echo './'`vorbis.c
```

Reply:
It's GCC correctly adding `-ansi`.

Here are problematic lines, personally I think dr_mp3 define tests are faulty here, but I haven't investigated it in detail:

https://github.com/dosbox-staging/dosbox-staging/blob/master/src/libs/decoders/dr_mp3.h#L229-L239

I think there's some misunderstanding in that comment.